providers/implementations/signature/sm2sig.c

```diff
@@ -108,8 +102,8 @@ static void *sm2sig_newctx(void *provctx, const char *propq)
         ERR_raise(ERR_LIB_PROV, ERR_R_MALLOC_FAILURE);
         return NULL;
     }
-    /* don't allow to change MD, and in fact there is no such need */
-    ctx->flag_allow_md = 0;
+    ctx->mdsize = SM3_DIGEST_LENGTH;
+    strcpy(ctx->mdname, OSSL_DIGEST_NAME_SM3);
```

Comment:
Why do you do this?  You use `OSSL_DIGEST_NAME_SM3` directly in the fetch anyway...

Reply:
The code has now changed to use ctx->mdname so it is not hardcoded anymore. We of course still keep the default being SM3.